examples/bosd-payload-with-rust-bitcoin.rs

```diff
@@ -0,0 +1,108 @@
+use base64::{engine::general_purpose::STANDARD, Engine};
+
+use bitcoin::{
+    absolute::LockTime,
+    bip32::{IntoDerivationPath, Xpriv},
+    key::Secp256k1,
+    secp256k1::All,
+    transaction::Version,
+    Amount, Network, Psbt, Transaction, TxOut,
+};
+
+fn main() {
+    // This example shows the basic usage of the [`bitcoin_bosd::Descriptor`] struct
+    // which is responsible for parsing and transmuting BOSD payloads into Bitcoin [`ScriptBuf`] and [`Address`].
+
+    // Example of a OP_RETURN payload.
+    let message = String::from("Bitcoin BOSD");
+    let payload = [&[0u8; 1], message.as_bytes()].concat(); // The BOSD payload is simply the type tag followed by the cryptographic payload.
+
+    let opreturn_desc = bitcoin_bosd::Descriptor::from_bytes(payload.as_slice()).unwrap();
+
+    let op_return_payload_script = opreturn_desc.to_script();
+
+    // Here we use the is_op_return() method from rust-bitcoin to check if the script is indeed an op_return.
+    assert!(op_return_payload_script.is_op_return());
+    // [`Descriptor`]s can be derived to a Bitcoin [`Address`].
+
+    // This will result in a error since this specific one is an OP_RETURN.
+    let op_return_address_error = opreturn_desc.to_address(Network::Bitcoin);
+    assert_eq!(
+        op_return_address_error.err().unwrap(),
+        bitcoin_bosd::DescriptorError::InvalidAddressConversion(
+            bitcoin_bosd::DescriptorType::OpReturn
+        )
+    );
+
+    //  Example of a P2WPKH payload.
+    let payload: [u8; 20] = [
+        // random hash160
+        0x97, 0xf5, 0xc9, 0x58, 0x7b, 0xb0, 0x5e, 0x3b, 0x59, 0x36, 0x16, 0x8d, 0x8b, 0x2a, 0xb2,
+        0x81, 0x94, 0x6e, 0xe3, 0xac,
+    ];
+    //  First byte is 3 to indicate a P2WPKH hash.
+    let payload = [&[3u8; 1], &payload[..]].concat();
+    let descriptor = bitcoin_bosd::Descriptor::from_bytes(payload.as_slice()).unwrap();
+
+    // An Anyone-can-pay transaction using the payloads.
+    let unsigned_tx = Transaction {
+        version: Version(2),
+        lock_time: LockTime::from_consensus(0),
+        input: vec![],
+        output: vec![
+            TxOut {
+                value: Amount::from_sat(0),
+                script_pubkey: descriptor.to_script(),
+            },
+            TxOut {
+                value: Amount::from_sat(0),
+                script_pubkey: op_return_payload_script,
+            },
+        ],
+    };
+    let mut psbt = Psbt::from_unsigned_tx(unsigned_tx).unwrap();
+    psbt.sign(&get_mock_key().0, &get_mock_key().1).unwrap();
+
+    #[cfg(feature = "examples-dep")]
+    // Encode the psbt transaction for base64 so we can check with analyzepsbt bitcoin-cli command.
+    println!("PSBT base64: {}", to_base64(&psbt.serialize()));
+}
+#[cfg(feature = "examples-dep")]
+/// Returns the base64 encoding of a serialized PSBT.
+fn to_base64(serialize: &[u8]) -> String {
+    STANDARD.encode(serialize)
+}
+/// Returns a derived key from a random master key.
```

Comment:
```suggestion
}

/// Returns a derived key from a random master key.
```
Empty line to separate things.